EIPS/eip-7377.md

```diff
@@ -0,0 +1,127 @@
+| `data`                 | `bytes`   |
+| `value`                | `int256`  |
+| `accessList`           | `List[Tuple[address, List[uint256]]]` |
+| `yParity`              | `uint8`   |
+| `v`                    | `uint256` |
+| `r`                    | `uint256` |
+
+The EIP-2718 `TransactionType` is `0x04` and the `TransactionPayload` is `rlp([chainId, nonce, maxFeePerGas, maxPriorityFeePerGas, gasLimit, codePtr, storageTuples, data, value, accessList, yParity, r, s])`.
+
+The transaction's signature hash is `keccak256(0x04 || rlp([chainId, nonce, maxFeePerGas, maxPriorityFeePerGas, gasLimit, codePtr, storageTuples, data, value, accessList])`
+
+#### Validation
+
+A MigrationTransaction is considered valid if the follow properties hold:
+
+* all [EIP-1559](./eip-1559.md) properties, unless specified otherwise
+* the code at `codePtr` is less than the [EIP-170](./eip-170) limit of `24576`
+* the code at `codePtr` must not have size `0`
+
+The intrinsic gas calculation modified from [EIP-1559](./eip-1559) to be `21000 + 16 * non-zero calldata bytes + 4 * zero calldata bytes + 1900 * access list storage key count + 2400 * access list address count + 15000 * length of storage`.
+
+#### Processing
+
+Execution a MigrationTransaction transaction has two parts.
+
+##### Contract Deployment
+
+Unlike standard contract deployment, the MigrationTransaction directly specifies what `code` value of the sender's account should be set to.
+
+As the first step of processing the transaction, set the sender's `code` value equal to `state[tx.codePtr].code`. Next, iterate through each tuple in `tx.storage` and in the sender's storage trie, set `storage[t.first] = t.second`.
+
+##### Transaction Execution
+
+Now instantiate an EVM call into the sender's account using the same rules as [EIP-1559](./eip-1559) and set the transaction's origin to be `keccak256(sender)[0..20]`.
+
+## Rationale
+
+### No `to` address field
+
+This transaction is only good for one time use to migrate an EOA to a smart contract. It is designed to immediately call the deployed contract, which is at the sender's address, after deployment to allow the sender to do any kind of further processing.
+
+### Code pointer for deployment
+
+Naively, one could design the migration transaction to have a field `code` that is just bytes. The outcome of this is that there would be substantial duplication of code calldata, since many users will want to deploy the exact same thing (a wallet). Using a pointer instead acknowledges the overwhelming use case for the transaction type, and exploits it as an optimization.
+
+### Cheaper storage
+
+Sometimes it's better to lead with a carrot instead of a stick.
+
+### Intrinsic does not account for contract deployment
+
+This is takes advantage of the fact that clients tend to single, unique copies of code. Therefore, the only operation here is changing a pointer in the state trie to desired code's pointer.
```

Comment:
```suggestion
Clients tend to single, unique copies of code. Therefore, the only operation clients have to perform is changing the account's code pointer in the state trie.
```
How about this? Also, there's a typo in your original phrasing ("This is takes...")

Reply:
```suggestion
This take advantage of the fact that clients tend to store a single, unique copy of code; no matter the number of deployments. Therefore, the only operation here is changing a pointer in the state trie to the desired code.
```